### PR TITLE
Make params optional argument

### PR DESCRIPTION
**Pull request by mohammad0-0ahmad**
For avoiding TS error without need to specify params type

### CODE REVIEW

File: src/makeStyles.tsx [hunk 1]
```diff
@@ -37,7 +37,7 @@ export function createMakeStyles<Theme>(params: { useTheme: () => Theme }) {
             cssObjectByRuleNameOrGetCssObjectByRuleName:
                 | ((
                       theme: Theme,
-                      params: Params,
+                      params?: Params,
```
Comment (garronej) [inline src/makeStyles.tsx:40]: https://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html#optional-parameters-in-callbacks

File: src/makeStyles.tsx [hunk 2]
```diff
@@ -56,7 +56,7 @@ export function createMakeStyles<Theme>(params: { useTheme: () => Theme }) {
 
             const outerCounter = getCounter();
 
-            return function useStyles(params: Params) {
+            return function useStyles(params?: Params) {
```
Comment (garronej) [inline src/makeStyles.tsx:59]: In recent TypeScript a function of type:
```typescript
declare function f(x: void);
```

Can be invoked like that:

```typescript
f();
```

[See compile time unit tests](https://github.com/garronej/tss-react/blob/63715700e16246ab3027ada4e712c738de3e9e2a/src/test/types/makeStyles.tsx#L44)

**Comment by garronej** [comment]
Hi @mohammad0-0ahmad, 
Thank you for the PR. 
I assume you are running an old typescript version.  
See review.  

I close but feel free to submit an example of problematic usecase and I'll reopen.  

Best

**Comment by mohammad0-0ahmad** [comment]
Hi @garronej !
Thanks for your quick response I will create new PR with link to an example repo.
I'm using TS v4.5.4
Best regards.

**Pull request closed without merging**